// react-rubiks-cube-utils/lib/DisplayCube.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
function DisplayCube({ cube, size }) {
    const colorMap = {
        W: "white",
        Y: "yellow",
        G: "green",
        B: "blue",
        R: "red",
        O: "orange",
    };
    return (react_1.default.createElement("div", { style: {
            width: "100%",
            height: "100%",
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
        } },
        react_1.default.createElement("div", { id: "cube-display", style: { display: "flex", flexDirection: "row", alignItems: "center" } },
            react_1.default.createElement("div", { id: "L", style: { flexDirection: "column", display: "flex", margin: "2px" } }, cube.L.map((row, rowIndex) => (react_1.default.createElement("div", { key: rowIndex, style: {
                    display: "flex",
                    flexDirection: "row",
                } }, row.map((color, colIndex) => (react_1.default.createElement("div", { key: colIndex, style: {
                    backgroundColor: colorMap[color]
                        ? colorMap[color]
                        : "black",
                    height: size,
                    width: size,
                    borderRadius: "2px",
                    margin: "2px",
                } }))))))),
            react_1.default.createElement("div", { style: {
                    display: "flex",
                    flexDirection: "column",
                    justifyContent: "space-between",
                    alignItems: "center",
                } },
                react_1.default.createElement("div", { id: "U", style: { flexDirection: "column", display: "flex", margin: "2px" } }, cube.U.map((row, rowIndex) => (react_1.default.createElement("div", { key: rowIndex, style: {
                        display: "flex",
                        flexDirection: "row",
                    } }, row.map((color, colIndex) => (react_1.default.createElement("div", { key: colIndex, style: {
                        backgroundColor: colorMap[color]
                            ? colorMap[color]
                            : "black",
                        height: size,
                        width: size,
                        borderRadius: "2px",
                        margin: "2px",
                    } }))))))),
                react_1.default.createElement("div", { id: "F", style: { flexDirection: "column", display: "flex", margin: "2px" } }, cube.F.map((row, rowIndex) => (react_1.default.createElement("div", { key: rowIndex, style: {
                        display: "flex",
                        flexDirection: "row",
                    } }, row.map((color, colIndex) => (react_1.default.createElement("div", { key: colIndex, style: {
                        backgroundColor: colorMap[color]
                            ? colorMap[color]
                            : "black",
                        height: size,
                        width: size,
                        borderRadius: "2px",
                        margin: "2px",
                    } }))))))),
                react_1.default.createElement("div", { id: "D", style: { flexDirection: "column", display: "flex", margin: "2px" } }, cube.D.map((row, rowIndex) => (react_1.default.createElement("div", { key: rowIndex, style: {
                        display: "flex",
                        flexDirection: "row",
                    } }, row.map((color, colIndex) => (react_1.default.createElement("div", { key: colIndex, style: {
                        backgroundColor: colorMap[color]
                            ? colorMap[color]
                            : "black",
                        height: size,
                        width: size,
                        borderRadius: "2px",
                        margin: "2px",
                    } })))))))),
            react_1.default.createElement("div", { id: "R", style: { flexDirection: "column", display: "flex", margin: "2px" } }, cube.R.map((row, rowIndex) => (react_1.default.createElement("div", { key: rowIndex, style: {
                    display: "flex",
                    flexDirection: "row",
                } }, row.map((color, colIndex) => (react_1.default.createElement("div", { key: colIndex, style: {
                    backgroundColor: colorMap[color]
                        ? colorMap[color]
                        : "black",
                    height: size,
                    width: size,
                    borderRadius: "2px",
                    margin: "2px",
                } }))))))),
            react_1.default.createElement("div", { id: "B", style: { flexDirection: "column", display: "flex", margin: "2px" } }, cube.B.map((row, rowIndex) => (react_1.default.createElement("div", { key: rowIndex, style: {
                    display: "flex",
                    flexDirection: "row",
                } }, row.map((color, colIndex) => (react_1.default.createElement("div", { key: colIndex, style: {
                    backgroundColor: colorMap[color]
                        ? colorMap[color]
                        : "black",
                    height: size,
                    width: size,
                    borderRadius: "2px",
                    margin: "2px",
                } }))))))))));
}
exports.default = DisplayCube;
